components/widgets/inputs/url_input/url_input.tsx

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import classNames from 'classnames';
+
+import OverlayTrigger from 'components/overlay_trigger';
+import Tooltip from 'components/tooltip';
+import Input from '../input/input';
+
+import Constants from 'utils/constants';
+import {getShortenedURL} from 'utils/url';
+
+import './url_input.scss';
+
+type URLInputProps = {
+    base: string;
+    path?: string;
+    pathInfo: string;
+    limit?: number;
+    maxLength?: number;
+    shortenLength?: number;
+    error?: string;
+    className?: string;
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    onBlur?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+function UrlInput({
+    base,
+    path,
+    pathInfo,
+    limit,
+    maxLength,
+    shortenLength,
+    error,
+    className,
+    onChange,
+    onBlur,
+}: URLInputProps) {
+    const {formatMessage} = useIntl();
+
+    const [editing, setEditing] = useState(false);
+
+    useEffect(() => {
+        if (error) {
+            setEditing(true);
+        }
+    }, [error]);
+
+    const fullPath = `${base}/${path ? `${path}/` : ''}`;
+    const fullURL = `${fullPath}${editing ? '' : pathInfo}`;
+    const isShortenedURL = shortenLength && fullURL.length > shortenLength;
+    const hasError = Boolean(error);
+
+    const handleOnInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+
+        if (onChange) {
+            onChange(event);
+        }
+    };
+
+    const handleOnInputBlur = (event: React.ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+
+        setEditing(hasError);
+
+        if (onBlur) {
+            onBlur(event);
+        }
+    };
+
+    const handleOnButtonClick = () => {
+        if (!hasError) {
+            setEditing(!editing);
+        }
+    };
+
+    const urlInputLabel = (
+        <span className='url-input-label'>
+            {formatMessage({id: 'url_input.label.url', defaultMessage: 'URL: '})}
+            {isShortenedURL ? getShortenedURL(fullURL, shortenLength) : fullURL}
```

Comment:
Saw a usage up above that appended a `/` at the end of the value. Is that also needed here?

Reply:
Not needed in here, `fullURL` will be already set.